graphene_federation/utils.py

```diff
@@ -26,3 +27,38 @@ def type_attribute_to_field_name(schema: Schema) -> Callable[[str], str]:
         return lambda attr_name: to_camel_case(attr_name)
     else:
         return lambda attr_name: attr_name
+
+
+def check_fields_exist_on_type(fields: set, type_: ObjectType):
+    return fields.issubset(set(type_._meta.fields))
+
+
+def is_valid_compound_key(type_name: str, key: str, schema: Schema):
```

Comment:
@patrick91 @erikwrede @arun-sureshkumar 
For validating compound keys, I have implemented a hacky way here in is_valid_compound_key(). Do you have any better solutions/suggestions?

Reply:
the way I'd do it is by parsing the key and then finding the fields in the types

Reply:
@patrick91 @erikwrede Please review the new implementation of is_valid_compound_key. Please suggest if there is any better method to parse the compound key.

Reply:
I mean using GraphQL core's parsing capabilities: https://graphql-core-3.readthedocs.io/en/latest/usage/parser.html

Reply:
I missed that initially 😅. Please review the current implementation using the graphql-core parse()